prime-router/src/main/kotlin/Translator.kt

```diff
@@ -97,23 +93,20 @@ class Translator(private val metadata: Metadata, private val settings: SettingsP
                 ?: error("qualityFilter $fnName is not found in list of JurisdictionalFilters")
             Pair(filterFn, fnArgs)
         }
-        val qualityFilteredReport = jurisFilteredReport.filter(
+        val qualityFilteredReport = input.filter(
             qualityFilterAndArgs,
             receiver,
             isQualityFilter = true,
             receiver.reverseTheQualityFilter
         )
-        if (qualityFilteredReport.itemCount != jurisFilteredReport.itemCount) {
+        if (qualityFilteredReport.itemCount != input.itemCount) {
             logger.warn(
                 "Data quality problem in report ${input.id}, receiver ${receiver.fullName}: " +
-                    "There were ${jurisFilteredReport.itemCount} rows prior to qualityFilter, and " +
+                    "There were ${input.itemCount} rows prior to qualityFilter, and " +
                     "${qualityFilteredReport.itemCount} rows after qualityFilter."
             )
         }
 
-        // Always succeed in translating an empty report after filtering (even if the mapping process would fail)
-        if (qualityFilteredReport.isEmpty()) return buildEmptyReport(receiver, input)
```

Comment:
Is this logic still needed?

Reply:
I was thinking no because it's equivalent (since it's `isEmpty`) but now I'm pretty sure it is.. will add the early return back.